### DISCUSSION TITLE
Cannot refactor "myclassname".java that is defined outside of an open project.

### DISCUSSION THREAD

**Discussion by mpet** [apache/netbeans#6035] [General]
Hi,

I have a java maven project with modules.
I start with building the project using maven.

Goals: clean install -DskipTests

It is successful!

Then I select, in "Files"  my abstract class name in a module and right click and
select "Refactor" and then "Rename".
Then I get a pop-up saying:

"Cannot refactor MDCCallable.java that is defined outside an open project."

I also tried using the "CTRL"+R and with the same result.

This always fails.

![image](https://github.com/apache/netbeans/assets/1445826/c22a027c-ca00-4acf-b9a8-d88e555eda44)

Anyone else that has the same problem for maven and java projects?

//mike

**Comment by mbien** [comment]
well... is the project open? :)

focus the file in the editor, then hit `ctrl+shift+1`, this should open the project/module the file is in.

**Comment by mpet** [comment]
@mbien  yes it works. It is not obvious to see if a project is open or closed. How do I know?

**Reply by mbien** [reply]
It is open if you see it in the "Projects" tab.

**Comment by mpet** [comment]
@mbien  that was quick :-) I guess I need to open each module for a maven project?
